homeassistant/components/alarmdecoder/alarm_control_panel.py

```diff
@@ -125,6 +125,10 @@ def _message_callback(self, message):
             "programming_mode": message.programming_mode,
             "ready": message.ready,
             "zone_bypassed": message.zone_bypassed,
+            "mask": message.mask,
+            "alarm_sounding": message.alarm_sounding,
+            "fire_alarm": message.fire_alarm,
+            "panel_type": message.panel_type
```

Comment:
Please explain what these attributes represent
Some of them could be eg binary sensors instead, perhaps?

Reply:
Sounds like the integration should be refactored to represent different partitions as different entities.

The fire alarm and alarm sounding reports should be binary sensors, if I interpret them correctly.

I don't find any info on panel type. What is that and why is it needed as a state attribute?

Reply:
The changes in the PR are geared towards a short term solution to provide users with a mechanism for working with multiple partitions.

Longer term the alarm decoder library would require changes to facilitate a refactoring of the HA integration. 

- mask - having an entity per partition this attribute would not be required but could remain as metadata for troubleshooting / debugging purposes.

- fire / alarm sounding - a binary sensor is ideal . Exposing the attributes now is also more of a short term solution. Having an entity per partition would require refactoring of some of the existing attributes as well as they are not partition specific, but applicable to the alarm system.

Current Attributes:
```
code_format: number
changed_by: null
code_arm_required: false
ac_power: true
alarm_event_occurred: false
backlight_on: false
battery_low: false
check_zone: false
chime: false
entry_delay_off: false
programming_mode: false
ready: false
zone_bypassed: false
mask: 35088
alarm_sounding: false
fire_alarm: false
panel_type: 0
friendly_name: Alarm Panel
supported_features: 7
```
- panel_type - the idea was to expose more panel metadata. In hindsight, not sure how this would be used. At a minimum if included, it should probably return the literal value. It appears to distinguish between ADEMCO/DSC panels.

Reply:
None of these attributes are in line with our entity design rules. We shouldn't add any of them as a short term solution.

The solution is to expose the values as separate entities.

Reply:
Without the 'mask' exposed, this integration is almost unusable for a system with multiple partitions. The Alarm Decoder repo (not HA integration) does not appear to have much activity which is concerning.

I'll look into making changes into the Alarm Decoder api, and maybe eventually more on the HA integration.

I'll just close this PR. Thanks Martin!